docs/snippets/numpydocExample.py

```diff
@@ -0,0 +1,323 @@
+    """
+    if param1 == param2:
+        raise ValueError('param1 may not be equal to param2')
+    return True
+
+
+def exampleGenerator(n):
+    """Generate an increasing sequence of numbers from 0 to a given limit.
+
+    Generators have a ``Yields`` section instead of a ``Returns`` section.
+
+    Parameters
+    ----------
+    n : `int`
+        The upper limit of the range to generate, from 0 to ``n`` - 1.
+
+    Yields
+    ------
+    number : `int`
+        The next number in the range of 0 to ``n`` - 1.
+
+    See also
+    --------
+    moduleLevelFunction
+
+    Examples
+    --------
+    Examples should be written in doctest format, and should illustrate how to
+    use the function:
+
+    >>> print([i for i in example_generator(4)])
+    [0, 1, 2, 3]
+    """
+    for i in range(n):
+        yield i
+
+
+class ExampleClass(object):
+    """Example class.
+
+    Parameters
+    ----------
+    param1 : `str`
+        Description of ``param1``.
+    param2 : `list` of `str`
+        Description of ``param2``.
+    param3 : `int`, optional
+        Description of ``param3``.
+    """
+
+    attr1 = None
+    """Description of ``attr1`` (`str`).
+    """
+
+    attr2 = None
+    """Description of ``attr2`` (`list` of `str`).
+    """
+
+    attr3 = None
+    """Description of ``attr3`` (`int`).
+    """
+
+    attr4 = None
+    """Description of ``attr4`` (`list` of `str`).
+    """
+
+    def __init__(self, param1, param2, param3):
+        self.attr1 = param1
+        self.attr2 = param2
+        self.attr3 = param3
+
+        self.attr4 = ['attr4']
+
+    @property
+    def readonlyProperty(self):
+        """Properties are documented in their getter method (`str`, read-only).
+        """
+        return 'readonlyProperty'
+
+    @property
+    def readwriteProperty(self):
+        """Properties with both a getter and setter are documented in their
+        getter method (`list` of `str`).
+
+        If the setter method contains notable behavior, it should be mentioned
+        here as well.
+        """
+        return self.attr4
+
+    @readwriteProperty.setter
+    def readwriteProperty(self, value):
+        self.attr4 = value
+
+    def exampleMethod(self, param1, param2):
+        """Test that a situation is true.
+
+        Class methods are similar to regular functions. Always use the
+        imperative mood when writing the one-sentence summary of a method or
+        function.
+
+        Parameters
+        ----------
+        param1 : obj
+            The first parameter.
+        param2 : obj
+            The second parameter.
+
+        Returns
+        -------
+        success : `bool`
+            `True` if successful, `False` otherwise.
+
+        Notes
+        -----
+        Do not include the ``self`` parameter in the ``Parameters`` section.
+        """
+        return True
+
+    def __special__(self):
+        """At the moment, special members with docstrings are not published in
```

Comment:
I hope this is going to be fixed; it seems like a substantial limitation.  Nevertheless, we should make clear that such members _must_ be documented anyway.